### ISSUE TITLE
Need to return ISO format datetime from get_forecasts service.

### ISSUE THREAD

**Issue by jkfranks9** [mawinkler/astroweather#60]
I have a script that parses the result from the weather integration `get_forecasts` (hourly) service into a list of items for a given set of hours. This works fine when the weather service is OpenWeatherMap, but fails for AstroWeather, because the parsed result is a string instead of a list. For example OpenweatherMap shows the following (I'm only including the first item for brevity). This is the response variable returned from `get_forecasts`, from the script trace:

```
  weather.openweathermap:
    forecast:
      - datetime: '2024-08-29T17:00:00+00:00'
        condition: sunny
        temperature: 93
        pressure: 30.09
        cloud_coverage: 3
        wind_speed: 4.14
        wind_bearing: 228
        uv_index: 8.34
        precipitation_probability: 0
        precipitation: 0
        apparent_temperature: 106
        dew_point: 76
        wind_gust_speed: 4.43
        humidity: 58
```

But AstroWeather shows:

```
  {'weather.astroweather_<name>': {'forecast': [{'datetime':
  datetime.datetime(2024, 8, 29, 17, 0, tzinfo=datetime.timezone.utc),
  'calm_percentage': 93, 'cloud_area_fraction_high': 0,
  'cloud_area_fraction_low': 0, 'cloud_area_fraction_medium': 13,
  'cloud_area_fraction': 14, 'cloudcover_percentage': 14,
  'cloudless_percentage': 86, 'condition': 76, 'fog_area_fraction': 0,
  'lifted_index': -4, 'precipitation_amount': 0.0, 'precipitation_probability':
  None, 'precipitation': None, 'seeing_percentage': 75, 'seeing': 0.62,
  'temperature': 95, 'transparency_percentage': 55, 'transparency': 0.45,
  'wind_bearing': 'SW', 'wind_speed': 1.1, 'humidity': 39}
```

The first part of the returned JSON result for OpenWeatherMap shows this (I modified my script to log the response variable):

```
{'forecast': [{'datetime': '2024-08-29T17:00:00+00:00', 'condition': 'sunny', ...
```

But Astroweather shows:

```
{'forecast': [{'datetime': datetime.datetime(2024, 8, 29, 17, 0, tzinfo=datetime.timezone.utc), 'calm_percentage': 93, ...
```

In checking the OpenWeatherMap code, I see that they use `isoformat()` to return the datetime as a string, but AstroWeather returns a datetime object.

When I add the  `isoformat()` call to the `_forecast` method in `weather.py` then all is well.

```
        for forecast in self.fcst_coordinator.data:
            forecasts.append(
                {
                    ATTR_FORECAST_TIME: forecast.forecast_time.isoformat(),
```

Note that I tried several different methods to deal with the AstroWeather result in my script but I could never manage to parse the result as a list. So I think using `isoformat()` here is the correct solution.

**Comment by mawinkler** [comment]
You're correct; I'm returning a datetime, but it should still work within your script.

This is what I quickly tested:

```
sequence:
  - action: weather.get_forecasts
    data:
      type: hourly
    target:
      entity_id: weather.astroweather_backyard
    response_variable: conditions
  - action: input_text.set_value
    metadata: {}
    data:
      value: "{{ conditions['weather.astroweather_backyard'].forecast[0].datetime }}"
    target:
      entity_id: input_text.temp
```

The input_text.temp is set correctly to `2024-09-03 13:00:00+00:00`.

Am I missing something?

Sorry for late reply, I was on holiday.

**Comment by mawinkler** [comment]
I made the change in the weather entity and pushed it to main. No new release yet. Can you possibly test if it works for you now?

**Comment by jkfranks9** [comment]
Tested, and works perfectly. Thanks!

**Comment by mawinkler** [comment]
Solved with release [v0.60.0](https://github.com/mawinkler/astroweather/releases/tag/v0.60.0).
